### PR TITLE
feat(python, rust): respect column stats collection configurations

### PR DESCRIPTION
**Pull request by ion-elgreco**
# Description
All of the Rust and Python write actions will now properly adhere to the configuration regarding the amount of columns stats have to be collected for. Either by dataSkippingNumIndexedCols or dataSkippingStatsColumns.

# Related Issue(s)
- closes https://github.com/delta-io/delta-rs/issues/2427

### CODE REVIEW

**Review by mightyshazam — APPROVED** [review]

File: crates/core/src/writer/stats.rs [hunk 1]
```diff
@@ -1,7 +1,3 @@
-use std::sync::Arc;
```
Comment (roeap) [inline crates/core/src/writer/stats.rs:1]: while still on nightly i think, `rustfmt` has an [opinion](https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#group_imports) - i.e. the prior ordering would be the idiomatic one.

File: crates/core/src/writer/stats.rs [hunk 2]
```diff
@@ -76,7 +85,21 @@ fn stats_from_file_metadata(
...
+        if num_indexed_cols == -1 {
+            schema_descriptor.num_columns()
+        } else if num_indexed_cols >= 0 {
+            min(num_indexed_cols as usize, schema_descriptor.num_columns())
+        } else {
+            return Err(DeltaWriterError::DeltaTable(DeltaTableError::Generic(
+                "delta.dataSkippingNumIndexedCols valid values are >=-1".to_string(),
+            )));
+        }
+    };
```
Comment (roeap) [inline crates/core/src/writer/stats.rs:100]: would it be a bit clearer if we just create the names or indexes of the columns we want to collect stats for here, rather then introducing additional branching below?

File: python/deltalake/writer.py [hunk 3]
```diff
@@ -264,6 +265,24 @@ def write_deltalake(
...
+                if str_indexed_cols is not None:
+                    num_indexed_cols = int(str_indexed_cols)
+            string_stats_cols = configuration.get("delta.dataSkippingStatsColumns")
+            if isinstance(string_stats_cols, str):
+                stats_cols = string_stats_cols.split(",")
+                if stats_cols and isinstance(stats_cols, list):
+                    if not all(isinstance(inner, str) for inner in stats_cols):
+                        raise ValueError(
+                            "dataSkippingStatsColumns needs to be a comma-separated list of column names in string format. Example: 'foo,bar,baz'"
+                        )
```
Comment (roeap) [inline python/deltalake/writer.py:285]: is there a way for us to isolate reading the configuration in the rust side. as kernel progresses we will likely see a lot of this move there, and this way we have to keep multiple code locations in sync.

File: crates/core/src/writer/stats.rs (hunk 2 above)
Comment (ion-elgreco) [reply crates/core/src/writer/stats.rs:100]: Let me check, we need to fetch the stats by idx, so I might be able to construct a vec of indexes :)

File: python/deltalake/writer.py (hunk 3 above)
Comment (ion-elgreco) [reply python/deltalake/writer.py:285]: I have added the exact same code in Rust. What I can do is move that into a pub fn, that I can call from python

**Review by roeap — APPROVED** [review]
👍

**Pull request merged**